src/components/ContractTable.tsx

```diff
@@ -0,0 +1,178 @@
+    (value) =>
+      fuzzysort
+        .go(value, contracts, { keys: ["header.name", "header.sig"] })
+        .map((x) => x.obj),
+    [contracts]
+  );
+
+  const handleOnSelect = (_, __, _selectedContracts) => {
+    setSelectedContracts(_selectedContracts);
+  };
+
+  const handleOnSelectAll = (_, _selectedContracts) => {
+    setSelectedContracts(_selectedContracts);
+  };
+
+  const removeSelected = () => {
+    setIsLoading(true);
+    const sigs = selectedContracts.map((contract) => contract.header.sig);
+    session.client.removeAbiRecords({ sigs }, (err, val) => {
+      setIsLoading(false);
+      if (err) return console.error(err);
+      removeContracts(selectedContracts);
+    });
+  };
+
+  const onChange = ({ target: { value } }) => {
+    setInput(value);
+    const _contracts = value ? filter(value) : contracts;
+    setFilteredContracts(_contracts);
+    setSelectedContracts(intersectionBy(selectedContracts, _contracts, "key"));
+  };
+
+  return (
+    <div>
+      <div style={{ display: "flex" }}>
+        <Input
+          value={input}
+          placeholder="Filter"
+          disabled={isLoading}
+          onChange={onChange}
+          style={{ marginBottom: "1em" }}
+          allowClear
+        />
+        <Button
+          danger
+          type="text"
+          disabled={selectedContracts.length === 0}
+          onClick={removeSelected}
+          style={{ marginLeft: "1em" }}
+        >
+          Remove Selected
+        </Button>
+        <Button
+          key="sync-button"
+          type="link"
+          icon={<SyncOutlined />}
+          disabled={isLoading}
+          onClick={fetchRecords}
+        >
+          Sync
+        </Button>
+      </div>
+      <Table
+        dataSource={filteredContracts}
+        tableLayout="fixed"
+        rowKey={(r) => r.header.name}
+        loading={{
+          spinning: isLoading,
+          tip: "Loading...",
+          indicator: <LoadingOutlined />,
+        }}
+        pagination={{
+          position: ["bottomCenter"],
+          pageSize: CONTRACTS_PER_PAGE,
+          defaultCurrent: 1,
+          showSizeChanger: false,
+        }}
+        rowSelection={{
+          type: "checkbox",
+          onSelect: handleOnSelect,
+          onSelectAll: handleOnSelectAll,
+          selectedRowKeys: selectedContracts.map(
+            (contract) => contract?.header?.name
+          ),
+        }}
+        expandable={{
+          expandedRowRender: (record) => (
+            <Table
+              columns={[
+                { title: "Name", dataIndex: "name", key: "name" },
+                { title: "Type", dataIndex: "typeName", key: "typeName" },
+              ]}
+              dataSource={record.params}
+              pagination={false}
+              rowKey={(r) => r.name}
+            />
+          ),
+          rowExpandable: (record) => record.params.length > 0,
+        }}
+      >
+        <Table.Column
+          title="Name"
```

Comment:
Please change to "Function Name"

Reply:
[eea4e25](https://github.com/GridPlus/lattice-manager/pull/227/commits/eea4e25af17755c69d536b5c8ad1bcc0b971f258)